src/Host.cs

```diff
@@ -24,9 +24,10 @@ public Host(Store store)
                 throw new ArgumentNullException(nameof(store));
             }
 
-            _store = store;
+            // Create a separate store for the host
+            _store = Interop.wasm_store_new(store.EngineHandle);
```

Comment:
The only place I don't fully understand is why `Host`'s parameters have `Store`. Ideally there shall be an engine, having `Store` to create a new engine looks inefficient.

At this point, I wonder if `Store` needs to be removed/encapsulated by `Host` (and `StoreBuilder` replaced by `Engine)?

Reply:
I could add back the `Engine` abstraction.  I agree that passing in a store just to get the internal engine here is confusing.

Let me do that.